### PR TITLE
add digital ocean deploy guide

### PR DESCRIPTION
**Pull request by happysalada**
### Summary of changes

add digital ocean deploy guide #441 

### Comments

- I'm guessing we should handle migrations and database backup and upload here (I haven't  gotten that far with docker swarm yet, so let me know if this is not needed, otherwise I will wait until I try and verify how it works)
- Maybe we should add how to handle logs here? Or is that out of scope? The problem is, personally, I still don't know how to handle logs. If you think it belongs here, let me know and I can add it when I have figured it out.

### CODE REVIEW

File: docs/guides/digital_ocean.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+- ssh into your droplet and start by upgrading it `apt-get upgrade`. It might be missing some security upgrades
+
+- create a directory for your configuration files for example `mkdir /etc/opt/app `
+(etc is the directory mainly used for storing config files on linux, opt is generally used for things not necessary to the OS). Then cd into that directory
+
+- copy over your `.env` files and your docker-compose.yml file
+on your droplet `mkdir config`
+on your computer
+`scp ./config/docker.env root@my_droplet_ip:/etc/opt/app/config`
```
Comment (happysalada) [inline docs/guides/digital_ocean.md]: I'm always using root here, I know this is debatable, let me know

File: docs/guides/digital_ocean.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- login with docker (needed in order to pull the image from dockerhub)
+`docker login` (then follow the instructions)
+
+- start a swarm on your droplet (don't forget to replace my_droplet_ip_address with your droplet's ip address)
+`docker swarm init --advertise-addr my_droplet_ip_address --listen-addr my_droplet_ip_address`
+
+- then deploy your stack with
+`docker stack deploy -c docker-compose.yml my_app_name`
+
+### upgrading
```
Comment (happysalada) [inline docs/guides/digital_ocean.md]: I am still not 100% sure that this is the recommended way to proceed. I am testing stuff with `docker stack deploy` here as well. Any advice here welcome

**Comment by bitwalker** [comment]
Hey FYI, I'm moving some stuff around in the docs, so you'll have to rebase these PRs, I should have the new docs up tomorrow. I should have time to review everything then as well.

**Comment by bitwalker** [comment]
@happysalada Can you migrate all of your doc PRs to the new documentation format and organization? If you have any questions, please let me know!

**Comment by happysalada** [comment]
@bitwalker one question. When you mean new documentation format, you mean changing the structure of that document to fit more the `deploying_to_aws` guide?
like having a `features`, a `prerequisite` and breaking down each step with `##`?
(just not sure about the format part)

**Comment by bitwalker** [comment]
I meant more that I want to be consistent with the folder organization, list style, capitalization in headers, and usage of tips/info/warning callouts - not that you aren't necessarily (I haven't been able to review yet), but I wanted to make sure that now that the docs are reworked, that you had a chance to tweak your changes if necessary. Hoping to review tomorrow

**Comment by happysalada** [comment]
I need to rewrite a bit then. I'll update when it's ready for review

File: docs/guides/deploying_to_digital_ocean.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Deploying with digital ocean
+
+There are multiple ways to deploy to digital ocean. We will deploy with docker swarm in this guide for simplicity.
+(note: while it's possible to use docker-compose it's generally not recommended for production. Since it's not much more difficult to use docker swarm, that's what we recommend.)
```
Comment (bitwalker) [inline docs/guides/deploying_to_digital_ocean.md]: I'm not sure what you mean in the last sentence here ("Since it's not much more difficult to use docker swarm, that's what we recommend"). Could you clarify what you mean?

Also, looking through the doc, you should make sure to capitalize all the pronouns (i.e. `digital ocean` should always be `Digital Ocean`, and `docker swarm` should always be `Docker Swarm`)

File: docs/guides/deploying_to_digital_ocean.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## single machine with docker
+
+### deploying
+
+Prerequisite:
```
Comment (bitwalker) [inline docs/guides/deploying_to_digital_ocean.md]: Prerequisites* (plural, not singular)

File: docs/guides/deploying_to_digital_ocean.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+- make sure you have completed the docker deployment guide
+- you know how to create a 'one-click app' droplet
+- you know how to ssh in a digital ocean droplet
+- you have published your image to dockerhub
+
+- Start by creating a new droplet. Click on the 'one-click apps' and choose docker. It will create a droplet for you with docker already installed
```
Comment (bitwalker) [inline docs/guides/deploying_to_digital_ocean.md]: No need to make each of the instructions here a list item, they should be paragraphs.

File: docs/guides/deploying_to_digital_ocean.md (hunk 3 above)
Comment (happysalada) [reply docs/guides/deploying_to_digital_ocean.md]: I'll leave out the last sentence then, it might be easier.
I originally added that sentence for somebody like. Docker-compose was just so easy, and I thought docker-swarm would be a lot heavier (a bit like kubernetes I guess).

Comment (bitwalker) [reply docs/guides/deploying_to_digital_ocean.md]: Ah I see, I would maybe rephrase this paragraph to:

While there are multiple ways one can deploy to Digital Ocean, for this guide we're choosing to use Docker Swarm (installed as a one-click app). This choice was made because it is easy to get started with on Digital Ocean, and makes for a good foundation for small production applications due to it's convenience and feature set.

**Comment by happysalada** [comment]
@bitwalker it took me a little longer than what I wanted to get back to this, sorry about that. Let me know if the changes made address everything. If yes, I'll squash my commits if you're okay with that.

**Comment by happysalada** [comment]
@bitwalker ready for another round of review

File: docs/guides/deploying_to_digital_ocean.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+Start a swarm on your droplet (don't forget to replace my_droplet_ip_address with your droplet's ip address) `docker swarm init --advertise-addr my_droplet_ip_address --listen-addr my_droplet_ip_address`
+
+Deploy your stack with `docker stack deploy -c docker-compose.yml my_app_name`
+
+### upgrading
+
+If you have tagged your image with `latest`, the chances are, when you want to upgrade to the next version of your app, it won't pull the new image. Here is how to force the upgrade
+
+- pull the latest image version `docker pull dockerhub_usersname/my_repo:latest`
```
Comment (wevtimoteo) [inline docs/guides/deploying_to_digital_ocean.md]: Fix typo `%s/usersname/username`

File: docs/guides/deploying_to_digital_ocean.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+- update your service `docker service update --image dockerhub_usersname/my_repo:latest --force my_service_name`. (to find out your service name, user `docker service ls`. Basically it will be something like my_app_name_web if you have followed exactly the docker deploy guide)
```
Comment (wevtimoteo) [inline docs/guides/deploying_to_digital_ocean.md]: Also here

**Comment by bitwalker** [comment]
Similar to the other PR, this has been merged manually in 1285010

Please review and give me any feedback you have, open new issues if there are things which need to be fixed.

Thanks again @happysalada!

**Comment by happysalada** [comment]
Looks good!

**Comment by bitwalker** [comment]
Awesome! Thanks @happysalada :)
